src/System.Text.Encodings.Web/tests/HtmlEncoderTests.cs

```diff
@@ -13,18 +13,22 @@ namespace Microsoft.Framework.WebEncoders
 {
     public class HtmlEncoderTests
     {
-        [Fact]
-        public void TestSurrogate()
+        [Theory]
+        [InlineData("&#x1F4A9;", "\U0001f4a9")]
+        [InlineData("&#x1F602;2", "\U0001F6022")]
+        [InlineData("&#x1F602; 21", "\U0001F602 21")]
+        [InlineData("x&#x1F602;y", "x\U0001F602y")]
```

Comment:
One more test please, to capture another boundary condition. This tests an even-length input being provided to the `EncodeCore` method, not just to the `Encode` method.

`[InlineData("&#x1F602;x&#x1F602;y", "\U0001F602x\U0001F602y")]`

Reply:
Added and committed my changes.